packages/headless/doc-parser/src/interface-resolver.ts

```diff
@@ -21,12 +21,70 @@ import {
   buildParamEntity,
 } from './entity-builder';
 
-export function resolveInterfaceMembers(
+export type InterfaceReferencesCount = Record<string, number>;
+
+export function resolveInterface(
+  entry: ApiEntryPoint,
+  apiInterface: ApiInterface,
+  referencesCount: InterfaceReferencesCount,
+  source?: ApiPropertySignature | Parameter
+) {
+  const typeName = apiInterface.name;
+  if (!referencesCount[typeName]) {
+    referencesCount[typeName] = 1;
+    const members = resolveInterfaceMembers(
+      entry,
+      apiInterface,
+      referencesCount
+    );
```

Comment:
Members are only parsed when an interface wasn't referenced before, this prevents an infinite recursive loop.